### PR TITLE
Feat - View Live Web Map!

### PR DESCRIPTION
**Pull request by sidsharma2002**
Screenshot_20220131-114932_Tree Tracker(debug)
Screenshot_20220131-114945_Tree Tracker(debug)

### CODE REVIEW

File: app/src/main/java/org/greenstand/android/TreeTracker/signup/CredentialEntryView.kt [hunk 1]
```diff
@@ -93,67 +98,109 @@ fun CredentialEntryView(viewModel: SignupViewModel, state: SignUpState) {
...
+            viewModel.updateCredentialType(Credential.Phone())
         }
+    )
+}
+
+@Composable
+fun RenderViewWebMapText() {
+    val context = LocalContext.current
+
+    val navigateToWebPage: () -> Unit = {
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/signup/CredentialEntryView.kt:144]: Can we move this function inline where it's called?

**Review by Elforama — CHANGES_REQUESTED** [review]
Looks great! I just added some minor comments to address if you can.

File: app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt [hunk 2]
```diff
@@ -84,7 +93,10 @@ class SignupViewModel(private val users: Users) : ViewModel() {
         _state.value = _state.value?.copy(credential = updatedCredential)
     }
 
-    fun doesCredentialExist() {
+    /**
+     *  update _state according to user existence
+     */
+    fun updateStateAccToUserExistence() {
```
[the remark this answers is not in the document]
Comment (sidsharma2002) [reply app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt:99]: ok, but earlier it was doesUserExists() which was totally out of context I think

File: app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt [hunk 3]
```diff
@@ -54,14 +55,22 @@ sealed class Credential {
...
+) : ViewModel() {
 
     private val _state = MutableLiveData(SignUpState())
     val state: LiveData<SignUpState> = _state
 
     fun updateName(name: String) {
         _state.value = _state.value?.copy(name = name)
     }
+
+    fun isInternetAvailable() : Boolean{
```
[the remark this answers is not in the document]
Comment (sidsharma2002) [reply app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt:70]: got it

File: app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (EngrTaofeek) [reply app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt:99]: I quite agree with @Elforama  view. Check out the interesting answers here about some naming conventions https://stackoverflow.com/questions/3874350/naming-conventions-for-java-methods-that-return-boolean

[the remark this answers is not in the document]
Comment (Elforama) [reply app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt:99]: To simplify it, we can name is 'submitInfo()' or something of that sort. From the views point of view, that's what is happening.

File: app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt [hunk 4]
```diff
@@ -54,14 +56,28 @@ sealed class Credential {
...
+class SignupViewModel(
+    private val users: Users,
+    private val checkForInternetUseCase: CheckForInternetUseCase
+) : ViewModel() {
 
     private val _state = MutableLiveData(SignUpState())
     val state: LiveData<SignUpState> = _state
 
+    private val _isInternetAvailable = MutableLiveData<Boolean>()
+    val isInternetAvailable: LiveData<Boolean> = _isInternetAvailable
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt:68]: This is my mistake, I should I mentioned this earlier.

For the each screen, we're trying to keep it simple by having them all rely on a single 'state' object called "state". In this case we can put the 'isInternetAvailable' as a field inside the SignupState object. This object is passed to the screen already, so you'd access it like 'state.isInternetAvailble'.

To see how the state object is updated, you can see the 'updateName' method.

File: app/src/main/java/org/greenstand/android/TreeTracker/view/TextButton.kt [hunk 5]
```diff
@@ -113,6 +116,107 @@ fun ApprovalButton(
     }
 }
 
+@Composable
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/view/TextButton.kt:119]: Please rebase to remove changes that have been merged already.

File: app/src/main/java/org/greenstand/android/TreeTracker/utilities/Constants.kt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package org.greenstand.android.TreeTracker.utilities
+
+object Constants {
+    const val TREE_TRACKER_URL = "http://www.treetracker.org"
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/utilities/Constants.kt:4]: This link doesn't seem to work for me. You can use "https://map.treetracker.org/"

**Review by Elforama — CHANGES_REQUESTED** [review]
Thanks for making the fixes. I've tested it out and it works well! I've added a few more comments, once they are addressed this should be good to merge.

**Pull request merged**